azure-pipelines/pull-request-validation/common.yml

```diff
@@ -50,7 +50,7 @@ jobs:
     inputs:
       javaHomeSelection: JDKVersion
       jdkVersionOption: "1.11"
-      tasks: common:testLocalDebugUnitTest -PuseMockApiForNativeAuth=true
+      tasks: common:testLocalDebugUnitTest
```

Comment:
Why are we disabling this flag from unit tests? We definitely need this for (some) of our unit tests.

On what build does `testLocalDebugUnitTest` run? If it's a release build, then you're right that setting `PuseMockApiForNativeAuth` would cause it to be set in the BuildConfig, which indeed we don't want. But if that's the case, then the solution isn't to disable the mock API altogether, but to create a test build. As we need to be able to continue to run the tests that need the mock API.

Reply:
This flag is not needed as we are enabling the mockAPIs for our test when needed. `NativeAuthControllerTest` is the only class in this test project that requires the mockAPI value to be true in BuildValues class. For `NativeAuthControllerTest` we are enabling the value before running any test and disabling once all tests have run.

Reply:
I'm not following. "When needed", is when, where? 

"For NativeAuthControllerTest we are enabling the value before running any test and disabling once all tests have run." - where is this done?

Reply:
Line 137-150 in `NativeAuthControllerTest.kt` file we are enabling the mock API flag for the duration of test run and then disabling it.